mail_forward/static/src/js/mail_forward.js

```diff
@@ -2,76 +2,122 @@
  * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
  */
 
-"use strict";
-openerp.mail_forward = function (instance) {
-    var _t = instance.web._t;
-    instance.mail.ThreadMessage.include({
-        bind_events: function () {
-            this._super.apply(this, arguments);
-            this.$('.oe_forward').on('click', this.on_message_forward);
-        },
+odoo.define('mail_forward.forward', function (require) {
+    "use strict";
+    var core = require('web.core');
+    var ChatThread = require('mail.ChatThread');
+    var chat_manager = require('mail.chat_manager');
+    var Model = require('web.Model');
+    var session = require('web.session');
+    var MessageModel = new Model('mail.message', session.user_context);
+    var _t = core._t;
 
-        on_message_forward: function () {
+    ChatThread.include({
+        on_message_forward: function (message_id) {
+            var self = this;
+            var read_fields = ['record_name', 'parent_id', 'subject', 'attachment_ids', 'date',
+                'email_to', 'email_from', 'email_cc', 'model', 'res_id', 'body'];
             // Generate email subject as possible from record_name and subject
-            var subject = [_t("FWD")];
-            if (this.record_name && (this.show_record_name ||
-                                     this.parent_id))
-            {
-                subject.push(this.record_name);
-            }
-            if (this.subject) {
-                subject.push(this.subject);
-            } else if (subject.length < 2) {
-                subject.push(_t("(No subject)"));
-            }
+            MessageModel.call('read', [message_id, read_fields], {context: session.user_context}).then(function (result) {
```

Comment:
Please put this into a separate, proxied, class-level method.

Just for inheritance/maintainability. :wink:

Reply:
Not sure what you mean by proxied method. I'm no javascript expert, I just ported the module to v10.